test/FixedStakingRewards.t.sol

```diff
@@ -998,4 +1074,420 @@ contract FixedStakingRewardsTest is Test {
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+    }
+
+    function test_Withdraw_RevertWhen_Paused() public {
+        // Set up staking first
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(2000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Release rewards and pause
+        stakingRewards.releaseRewards();
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.withdraw(50e18);
+        vm.stopPrank();
+    }
+
+    function test_GetReward_RevertWhen_Paused() public {
+        // Set up staking and rewards
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(1000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Move time forward and release rewards
+        skip(3600);
+        stakingRewards.releaseRewards();
+
+        // Pause the contract
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.getReward();
+        vm.stopPrank();
+    }
+
+    function test_Exit_RevertWhen_Paused() public {
+        // Set up staking first
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(1000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Release rewards and pause
+        stakingRewards.releaseRewards();
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.exit();
+        vm.stopPrank();
+    }
+
+    function test_OwnerFunctions_WorkWhen_Paused() public {
+        // Owner functions should still work when paused
+        stakingRewards.pause();
+
+        // These should all work
+        stakingRewards.addToWhitelist(user1);
+        stakingRewards.removeFromWhitelist(user1);
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(100e18);
+        stakingRewards.releaseRewards();
+
+        assertTrue(stakingRewards.paused());
+    }
+
+    function test_Integration_PauseUnpauseFlow() public {
```

Comment:
if we really need these types of integration tests, should we put them in a separate test file?